File: android/app/src/main/java/com/epam/kborodin/bored/data/local/dao/ActionDao.kt
package com.epam.kborodin.bored.data.local.dao

import androidx.room.*
import com.epam.kborodin.bored.data.local.entity.ActionEntity

@Dao
abstract class ActionDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    abstract suspend fun insertAction(entity: ActionEntity)

    @Query("SELECT * FROM `action`")
    abstract suspend fun getAction(): ActionEntity?

    @Query("DELETE FROM `action`")
    abstract suspend fun delete()

    @Transaction
    open suspend fun update(
        entity: ActionEntity,
    ) {
        delete()
        insertAction(entity)
    }
}

File: flutter_app/lib/data/local/dao/action_dao.dart
import 'package:floor/floor.dart';
import 'package:flutter_app/data/local/entity/action_entity.dart';

@dao
abstract class ActionDao {
  @insert
  Future<void> insertAction(ActionEntity entity);

  @Query('SELECT * FROM action')
  Future<ActionEntity?> getAction();

  @Query('DELETE FROM action')
  Future<void> deleteAction();

  @transaction
  Future<void> update(ActionEntity entity) async {
    await deleteAction();
    await insertAction(entity);
  }
}
